libs/evm2wasm/evm2wast.c

```diff
@@ -0,0 +1,689 @@
+				break;
+
+			case JUMPI:
+				has_jump=1;
+				append_segment("(set_local $jump_dest (call $check_overflow \
+                          (i64.load (get_global $sp)) \
+                          (i64.load (i32.add (get_global $sp) (i32.const 8))) \
+                          (i64.load (i32.add (get_global $sp) (i32.const 16))) \
+                          (i64.load (i32.add (get_global $sp) (i32.const 24))))) \
+                    (set_global $sp (i32.sub (get_global $sp) (i32.const 64))) \
+                    (br_if $loop (i32.eqz (i64.eqz (i64.or \
+                      (i64.load (i32.add (get_global $sp) (i 32.const 32))) \
+                      (i64.or \
+                        (i64.load (i32.add (get_global $sp) (i32.const 40))) \
+                        (i64.or \
+                          (i64.load (i32.add (get_global $sp) (i32.const 48))) \
+                          (i64.load (i32.add (get_global $sp) (i32.const 56))) \
+                        ) \
+                      ) \
+                    ))))");
+				add_stack_check(segment);
+				add_metering(&wast_code, &segment);
+				break;
+
+			case JUMPDEST:
+				end_segment(&wast_code, &segment);
+				jumps = safe_realloc(jumps, jumps_len);
+				jumps[jumps_len++] = i;
+				gas_count = 1;
+				break;
+			
+			case GAS:
+				append_segment("(call $GAS)\n");
+				add_metering(&wast_code, &segment);
+				break;
+
+			case LOG1:
+			case LOG2:
+			case LOG3:
+			case LOG4:
+			case LOG5:
+				format_segment("(call $LOG (i32.const %d))", 36, (int)evm_code[i]);
+				break;
+
+			case DUP1:
+			case DUP2:
+			case DUP3:
+			case DUP4:
+			case DUP5:
+			case DUP6:
+			case DUP7:
+			case DUP8:
+			case DUP9:
+			case DUP10:
+			case DUP11:
+			case DUP12:
+			case DUP13:
+			case DUP14:
+			case DUP15:
+			case DUP16:
+			case SWAP1:
+			case SWAP2:
+			case SWAP3:
+			case SWAP4:
+			case SWAP5:
+			case SWAP6:
+			case SWAP7:
+			case SWAP8:
+			case SWAP9:
+			case SWAP10:
+			case SWAP11:
+			case SWAP12:
+			case SWAP13:
+			case SWAP14:
+			case SWAP15:
+			case SWAP16:
+				format_segment("(call $%s (i32.const $%d))\n", 60,
+						op.name, (int)evm_code[i]);
+				break;
+
+			case PC:
+				format_segment("(call $PC (i32.const %ld", 40, i);
+				break;
+
+			case PUSH1:
+			case PUSH2:
+			case PUSH3:
+			case PUSH4:
+			case PUSH5:
+			case PUSH6:
+			case PUSH7:
+			case PUSH8:
+			case PUSH9:
+			case PUSH10:
+			case PUSH11:
+			case PUSH12:
+			case PUSH13:
+			case PUSH14:
+			case PUSH15:
+			case PUSH16:
+			case PUSH17:
+			case PUSH18:
+			case PUSH19:
+			case PUSH20:
+			case PUSH21:
+			case PUSH22:
+			case PUSH23:
+			case PUSH24:
+			case PUSH25:
+			case PUSH26:
+			case PUSH27:
+			case PUSH28:
+			case PUSH29:
+			case PUSH30:
+			case PUSH31:
+			case PUSH32:
+				{
+					char *bytes = padleft(evm_code+1, evm_code[i]-0x5f, 32);
+					i+=(size_t)evm_code[i]-0x5f;
+					if(!bytes) goto err;
+					int bytes_rounded = (int)ceil((double)(evm_code[i-1]/8));
```

Comment:
If you can, please avoid using float/double and math ops.

`(evm_code[i - 1] +7 ) / 8` should have the same effect.